nomos-blend/network/src/edge/behaviour.rs

```diff
@@ -0,0 +1,306 @@
+            peer_id,
+            handler: NotifyHandler::One(connection_id),
+            event: FromBehaviour::Message(message),
+        });
+        self.try_wake();
+        true
+    }
+
+    /// Schedules a [`EventToSwarm`] to be sent to the swarm.
+    fn schedule_event_to_swarm(&mut self, event: EventToSwarm) {
+        self.events.push_back(ToSwarm::GenerateEvent(event));
+        self.try_wake();
+    }
+
+    fn schedule_drop_substream(&mut self, peer_id: PeerId, connection_id: ConnectionId) {
+        tracing::debug!(target: LOG_TARGET, "Dropping substream: peer:{peer_id}, connection_id:{connection_id}");
+        self.events.push_back(ToSwarm::NotifyHandler {
+            peer_id,
+            handler: NotifyHandler::One(connection_id),
+            event: FromBehaviour::DropSubstream,
+        });
+        self.try_wake();
+    }
+
+    /// Handles [`ToBehaviour::ReadyToSend`] event from the connection handler.
+    fn handle_ready_to_send(&mut self, peer_id: PeerId, connection_id: ConnectionId) {
+        tracing::debug!(target: LOG_TARGET, "Conn is ready to send: peer:{peer_id}, connection_id:{connection_id}");
+        self.requested_dials.remove(&(peer_id, connection_id));
+        if !self.schedule_send_message(peer_id, connection_id) {
+            // There was no message to send. Drop the substream.
+            self.schedule_drop_substream(peer_id, connection_id);
+        }
+    }
+
+    /// Handles [`ToBehaviour::Dropped`] event from the connection handler.
+    fn handle_dropped_connection(
+        &mut self,
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        reason: Option<&FailureReason>,
+    ) {
+        self.requested_dials.remove(&(peer_id, connection_id));
+        if matches!(reason, Some(FailureReason::UpgradeError)) {
+            tracing::error!(target: LOG_TARGET, "Upgrade error: peer:{peer_id}, connection_id:{connection_id}");
+            if let Err(e) = self.schedule_dial_if_needed() {
+                tracing::error!(target: LOG_TARGET, "Failed to schedule dial: {e}");
+            }
+        }
+    }
+
+    /// Handles [`FromSwarm::DialFailure`] event from the swarm.
+    fn handle_dial_failure(&mut self, failure: DialFailure) {
+        tracing::error!(target: LOG_TARGET, "Dial failure: {failure:?}");
+        if let Some(peer_id) = failure.peer_id {
+            self.requested_dials
+                .remove(&(peer_id, failure.connection_id));
+        }
+        if let Err(e) = self.schedule_dial_if_needed() {
+            tracing::error!(target: LOG_TARGET, "Failed to schedule dial: {e}");
+        }
+    }
+
+    fn try_wake(&mut self) {
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
+        }
+    }
+
+    // TODO: Remove this method once the session stream is implemented.
+    //       https://github.com/logos-co/nomos/issues/1462
+    pub fn set_membership(&mut self, membership: Membership<PeerId>) {
```

Comment:
I think this is only used for tests, at least in this PR, right? Has this call been integrated into the libp2p stack in a different PR? Otherwise I'd mark this as a `test`-only function, if you agree.